actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -137,24 +137,24 @@ module FormOptionsHelper
       #
       # ==== Gotcha
       #
-      # The HTML specification says when +multiple+ parameter passed to select and all options got deselected
-      # web browsers do not send any value to server. Unfortunately this introduces a gotcha:
-      # if a +User+ model has many +roles+ and have +role_ids+ accessor, and in the form that edits roles of the user
-      # the user deselects all roles from +role_ids+ multiple select box, no +role_ids+ parameter is sent. So,
+      # The HTML specification says when the +multiple+ attribute is specified on a select element and all options
+      # are deselected, web browsers do not send any value to the server. Unfortunately, this introduces a gotcha:
+      # if a +User+ model has many +roles+ and a +role_ids+ accessor, and in the form that edits roles of the user
+      # the user deselects all roles from the +role_ids+ multiple select box, no +role_ids+ parameter is sent. So,
```

Comment:
While we are changing this, I think it would nice to show example code for the gotcha.
Maybe something like:
```suggestion
      # are deselected, web browsers do not send any value to the server. Unfortunately, this introduces a gotcha. For example, given a +User+ model that has many +roles+:
      #
      #   class User < ActiveRecord::Base
      #     has_many :roles
      #   end
      #
      #  and a multiple select in the form that edits roles of the user:
      #
      #    select :user, :role_ids, Role.all.collect { |r| [ r.name, r.id ] }, {}, { multiple: true })
      #
      # If all roles are deselected, no +role_ids+ parameter is sent. So,
```

Reply:
Or maybe keep with the Post example and just show the multi select:
```suggestion
      # are deselected, web browsers do not send any value to the server. Unfortunately, this introduces a gotcha. For example, given a multiple select in a form that edits roles of the user:
      #
      #   <select name="post[person_ids]" id="post_person_ids" multiple="true">
      #     <option value="" label=" "></option>
      #     <option value="1">David</option>
      #     <option value="2">Eileen</option>
      #     <option value="3">Rafael</option>
      #   </select>
      #
      #
      # If no persons are selected, no +person_ids+ parameter is sent. So,
```

Reply:
Keeping with the Post example is nice. So is showing example code of the helper. I just pushed an update that incorporates aspects from both suggestions. Let me know what you think. Thanks!

Reply:
Looks great!